crates/manifest/src/schema/v2.rs

```diff
@@ -265,6 +309,92 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    fn deserializing_labels() {
+        AppManifest::deserialize(toml! {
+            spin_manifest_version = 2
+            [application]
+            name = "trigger-configs"
+            [[trigger.fake]]
+            something = "something else"
+            [component.fake]
+            source = "dummy"
+            key_value_stores = ["default", "snake_case", "kebab-case"]
+            sqlite_databases = ["default", "snake_case", "kebab-case"]
+        })
+        .unwrap();
+    }
+
+    #[test]
+    fn deserializing_labels_fails_for_non_kebab_or_snake() {
+        assert!(AppManifest::deserialize(toml! {
+            spin_manifest_version = 2
+            [application]
+            name = "trigger-configs"
+            [[trigger.fake]]
+            something = "something else"
+            [component.fake]
+            source = "dummy"
+            key_value_stores = ["b@dlabel"]
+        })
+        .is_err());
+    }
+
+    fn get_test_manifest_with_labels(labels: Vec<String>) -> AppManifest {
+        AppManifest {
+            spin_manifest_version: FixedVersion::try_from(2).unwrap(),
+            application: AppDetails {
+                name: "trigger-configs".to_string(),
+                version: "".to_string(),
+                description: "".to_string(),
+                authors: vec![],
+                trigger_global_configs: Map::new(),
+                tool: Map::new(),
+            },
+            variables: Map::new(),
+            triggers: Map::new(),
+            components: {
+                let mut components = Map::new();
+                components.insert(
+                    KebabId::try_from("fake".to_owned()).unwrap(),
+                    Component {
+                        source: ComponentSource::Local("dummy".to_string()),
+                        description: "".to_string(),
+                        variables: Map::new(),
+                        environment: Map::new(),
+                        files: vec![],
+                        exclude_files: vec![],
+                        allowed_http_hosts: vec![],
+                        allowed_outbound_hosts: vec![],
+                        key_value_stores: labels.clone(),
+                        sqlite_databases: labels,
+                        ai_models: vec![],
+                        build: None,
+                        tool: Map::new(),
+                    },
+                );
+                components
+            },
+        }
+    }
+
+    #[test]
+    fn serialize_labels() {
+        let manifest = get_test_manifest_with_labels(vec![
+            "default".to_string(),
+            "snake_case".to_string(),
+            "kebab-case".to_string(),
+        ]);
+        let serialized = toml::to_string(&manifest).unwrap();
+        toml::from_str::<AppManifest>(&serialized).unwrap();
```

Comment:
Is it worth confirming here that the deserialised labels are what you expect?  I guess, as you say, that other tests can be a backstop for this...